### PR TITLE
Use deprecation logger for CLDR date format specifiers

### PR DESCRIPTION
**Pull request by thecoop**
Use the deprecation logger so warnings end up in Kibana.
Update regex for a few more textual specifiers.

I've had to modify or mute several tests to deal with the extra deprecation warning this outputs

### CODE REVIEW

**Review by mosche — APPROVED** [review]
lgtm 👍

File: server/src/main/java/org/elasticsearch/common/time/DateUtils.java [hunk 1]
```diff
@@ -391,13 +390,17 @@ public static ZonedDateTime nowWithMillisResolution(Clock clock) {
 
     // check for all textual fields, and localized zone offset
     private static final Predicate<String> CONTAINS_CHANGING_TEXT_SPECIFIERS = System.getProperty("java.locale.providers", "")
-        .contains("COMPAT") ? Pattern.compile("[EcGaO]|MMM|LLL|eee|ccc|QQQ|ZZZZ").asPredicate() : Predicates.never();
+        .contains("COMPAT") ? Pattern.compile("[BEGOavz]|LLL|MMM|QQQ|ccc|eee|Z{4}").asPredicate() : Predicates.never();
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/time/DateUtils.java:393]: What is `B`? Should `v` be upper case (time-zone ID)?

Comment (rjernst) [reply server/src/main/java/org/elasticsearch/common/time/DateUtils.java:393]: Can we have tests with these formats in DateUtilsTests to ensure we are checking/interpreting them correctly?

Comment (thecoop) [reply server/src/main/java/org/elasticsearch/common/time/DateUtils.java:393]: `B` is 'Period of day' ('in the morning'). `v` is generic time-zone name (`V` is time-zone id), which I've seen can be localised

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by thecoop** [comment]
@elasticmachine rerun elasticsearch-ci/7.17.25

**Comment by thecoop** [comment]
@elasticmachine update branch

**Review by rjernst — APPROVED** [review]
Still LGTM

**Pull request merged**